### PR TITLE
docs: use Helm v3 commands in Istio docs

### PR DESCRIPTION
**Pull request by masonlr**
**What this PR does / why we need it**:

Within the Istio docs, this PR:

1. Updates instances of `helm install` in the docs to be compatible with Helm v3 (the `--name foo` option for Helm release names is no longer supported in Helm v3 and the release name is now a positional argument).
2. Removes a broken link to `./examples/istio_canary.html`
3. Regenerates the Istio Jupyter notebook to remove minor errors that are shown here: https://docs.seldon.io/projects/seldon-core/en/v1.6.0/examples/istio_examples.html (the outputs for cells `[1]` and `[4]`)

@cliveseldon, for reference the original discussion on Slack was here: https://seldondev.slack.com/archives/CSVSD5S75/p1610047087283500?thread_ts=1609611010.277400&cid=CSVSD5S75

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

N/A

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @masonlr. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **sachinvarghese**
You can assign the PR to them by writing `/assign @sachinvarghese` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: notebooks/istio_example.ipynb
Comment (cliveseldon) [inline notebooks/istio_example.ipynb:32]: Do we need this notebook change added or is it an artifact of your testing?

**Review by cliveseldon — CHANGES_REQUESTED** [review]

File: notebooks/istio_example.ipynb
Comment (masonlr) [reply notebooks/istio_example.ipynb:32]: `"seldonio/mock_classifier:1.7.0-dev"` is now returned in the Rest and gRPC Response objects: i.e.  `meta.requestPath.model` is now present where previously `meta` was an empty object. When glancing the docs, it might be useful to see that this metadata exists.

Existing notebook with empty metadata:

image


New notebook with Response object metadata filled:

image



Your call though. The notebook doesn't necessarily need to be changed: the changes here update the minor existing error messages in the cell outputs – the cell inputs haven't been changed. Although they're minor, if you're skimming the docs for the first time and see errors, you might be concerned.

The minor existing error messages are just related to the idempotency of `kubectl create` (i.e. attempting to create an existing resource):

image

image

**Comment by masonlr** [comment]
I've added a new notebook for "Canary Rollout with Istio and Ambassador" following a Slack discussion with @axsaucedo (https://seldondev.slack.com/archives/C8Y9A8G0Y/p1615542087045500?thread_ts=1615537347.044200&cid=C8Y9A8G0Y).

**Comment by masonlr** [comment]
@axsaucedo, this page https://docs.seldon.io/projects/seldon-core/en/latest/examples/seldon_core_setup.html#Istio links to https://docs.seldon.io/projects/seldon-core/en/latest/workflow/install.html#install-istio-ingress-gateway.

One issue is that the heading `#install-istio-ingress-gateway` doesn't exist, making it difficult to find information when you are reading the docs for the first time. I'll update it to https://docs.seldon.io/projects/seldon-core/en/latest/workflow/install.html#ingress-support.

There are two concepts:

1. Installing istio itself – which is covered indirectly here https://github.com/SeldonIO/seldon-core/tree/master/examples/auth and hasn't made it into the docs;
2. Installing seldon core with a configuration that it is compatible with istio – which is covered here https://docs.seldon.io/projects/seldon-core/en/latest/ingress/istio.html.

**Comment by axsaucedo** [comment]
Cool - looks good, @cliveseldon are there any further changes? I wonder whether we'll ahve to trigger the notebook tests just to make sure all passes?

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[cliveseldon](https://github.com/SeldonIO/seldon-core/pull/3044#issuecomment-811772874 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
